packs/boundary_worker/templates/pack.nomad.tpl

```diff
@@ -0,0 +1,173 @@
+    restart {
+      [[- $restart_logic := var "nomad_group_restart_logic" . ]]
+      attempts = [[ $restart_logic.attempts ]]
+      interval = "[[ $restart_logic.interval ]]"
+      delay    = "[[ $restart_logic.delay ]]"
+      mode     = "[[ $restart_logic.mode ]]"
+    }
+
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/volume
+    [[/* iterate over `var.volumes` to create Volumes */]]
+    [[- range $index, $mount := var "nomad_group_volumes" . ]]
+    volume "[[ $mount.name ]]" {
+      source    = "[[ $mount.name ]]"
+      type      = "[[ $mount.type ]]"
+      read_only = [[ $mount.read_only ]]
+    }
+    [[ end ]]
+
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/task
+    task "[[ var "nomad_task_name" . ]]" {
+      # see https://developer.hashicorp.com/nomad/docs/drivers
+      driver = "[[ var "nomad_task_driver" . ]]"
+
+      # Action to print environment variables relevant to the Boundary Helper
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/action
+      action "print-env" {
+        command = "[[ var "app_boundary_helper_path" . ]]"
+
+        # see https://github.com/workloads/boundary-helper/#environment-variables
+        args = [
+          "env"
+        ]
+      }
+
+      # Action to clean up old workers based on "Last Seen" duration
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/action
+      action "cleanup-workers" {
+        command = "[[ var "app_boundary_helper_path" . ]]"
+
+        # see https://github.com/workloads/boundary-helper/#environment-variables
+        args = [
+          "cleanup-workers"
+        ]
+      }
+
+      # see https://developer.hashicorp.com/nomad/docs/drivers/raw_exec
+      # and https://developer.hashicorp.com/nomad/docs/drivers/exec
+      config {
+        command = "boundary"
+
+        args = [
+          "server",
+          "-config",
+          "${NOMAD_TASK_DIR}/config.hcl"
+        ]
+      }
+
+        # set `NOMAD_TOKEN` using Workload Identity
+      # see https://developer.hashicorp.com/nomad/docs/concepts/workload-identity
+      identity {
+        env = true
+      }
+
+      [[ template "environment_variables" . ]]
+
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/template
+      template {
+        change_mode = "restart"
+
+        data = [[ "<<DATA" ]]
+          [[ template "boundary_worker_config" . ]]
+        [[ "DATA" ]]
+
+        destination          = "${NOMAD_TASK_DIR}/config.hcl"
+        error_on_missing_key = true
+      }
+
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/volume_mount
+      [[/* iterate over `var.volumes` to create Volume Mounts */]]
+      [[- range $index, $mount := var "nomad_group_volumes" . ]]
+      volume_mount {
+        volume      = "[[ $mount.name ]]"
+        destination = "[[ $mount.destination ]]"
+        read_only   = [[ $mount.read_only ]]
+      }
+      [[ end ]]
```

Comment:
Is there anything that might be worth preserving or can the volume config be removed entirely?

Reply:
I'll take care of this in #23